src/edde/edde.py

```diff
@@ -10,35 +11,31 @@
     get_demographics_data,
     get_other_data,
 )
+from src.components.sidebar import branch_selectbox, output_selectbox
 from src.edde.components import column_comparison_table
 
 
 def edde():
     st.title("EDDE QAQC")
 
-    branches = get_active_s3_folders(
-        repo="db-equitable-development-tool", bucket_name="edm-publishing"
+    default_branch = get_default_branch()
+    branch = branch_selectbox(
+        repo="db-equitable-development-tool",
+        bucket_name="edm-publishing",
+        s3_folder="db-eddt",
+        label="Select a branch (will use latest)",
+        default=default_branch,
     )
-    branch = st.sidebar.selectbox(
-        "Select a branch (will use latest)",
-        branches,
-        index=branches.index("main"),
+    branch_for_comp = branch_selectbox(
+        repo="db-equitable-development-tool",
+        bucket_name="edm-publishing",
+        s3_folder="db-eddt",
+        label="Select a branch for comparison",
+        default=default_branch,
```

Comment:
So by trying to be more DRY I ended up repeating myself more here. But tough to think of a super elegant solution, this is a case where object-oriented is sort of nice - create some sort of interface that takes in repo, bucket, and can cache s3 folders, and then when producing a selectbox you don't need to re-enter all these things. You have any thoughts on how to make this a bit more elegant?

Reply:
It might make sense to have a "config" object somewhat akin to the DigitalOceanClient that doesn't have methods but just has stored fields. Basically a named tuple. That could have repo, bucket name, s3 folder, etc. Then we only have to declare once per data product/page and that gets passed around rather than sending along each field individually

Reply:
definitely like the idea of (at least) using a named tuple for a list of constants

I guess for this could be more DRY by setting the repo and bucket_name constants in `edde.helpers`, especially since I'm seeing the string `db-eddt` use a bunch of times in that file

Reply:
Oh good call I'll do that at least for now